tests/functional/tests/modules/test_cloudsupport.py

```diff
@@ -144,6 +144,20 @@ def test_20_create_instance(self):
     @retry(stop=stop_after_attempt(18), wait=wait_fixed(20))
     def test_25_test_connectivity(self):
         """Test: connectivity of an instance."""
+        # Workaround on CI environment that the DNS can not resolve the
+        # hostname of the dhcp_agent/hypervisor machine.
+        try:
+            # Is OVS
+            model.get_application("neutron-gateway")
+            app = "neutron-gateway"
+        except KeyError:
+            # Is OVN
+            app = "nova-compute"
+        host_ip = model.get_app_ips(app)[0]
+        host = model.run_on_unit(f"{app}/0", "hostname").get("Stdout").rstrip()
+        cmd = f"""echo "{host_ip} {host}" >> /etc/hosts """
```

Comment:
The double quotes here look a bit weird. Don't they need to be escaped maybe?

Reply:
Since the func tests are passing, maybe there's nothing wrong with the syntax then.